### DISCUSSION TITLE
Explanation of Buffer of DeviceClient

### DISCUSSION THREAD

**Discussion by haverboecker** [Azure/azure-iot-sdk-java#1627] [Q&A] [answered]
Hello everybody!

I have a question about the normal DeviceClient.
The docs state, that the DeviceClient will buffer messages when the network is down (I assume when no connecton to the IotHub is available) and once the connection comes back up, it will send them. Example here. https://azure.github.io/azure-iot-sdk-java/master/device/com/microsoft/azure/sdk/iot/device/DeviceClient.html 

I would like to know how this buffer works. Are there any docs or explanation for this?
Like: How long are messages buffered, is it possible to persist them on the filesystem, is there any way to configure the behaviour of the buffer?

I didn't find anything regardings buffers in the discussions. Thanks for any hints!

Best regards,

Peer

**Comment by timtay-microsoft** [comment, marked as answer]
Hi @haverboecker! The device client buffer persists in memory, not disk storage. There currently is no way to persist them on the file system instead. You can set an expiry time on each message before you enqueue it. If you don't, then the message can stay queued indefinitely.

**Reply by haverboecker** [reply]
OK, thanks!
